src/big/big_float.cr

```diff
@@ -93,11 +93,11 @@ struct BigFloat < Float
     LibGMP.mpf_cmp_z(self, other)
   end
 
-  def <=>(other : Float::Primitive)
+  def <=>(other : Float::Primitive) : Int32?
```

Comment:
All the `<=>` methods should normally just return `Int32`

Reply:
Agreed, and the `Nil` return is a bit confusing here as well (though not incorrect when considering you're comparing with a `NaN`).

Reply:
> All the <=> methods should normally just return Int32

The contract in general is to return `nil` when not comparable. That is relevant for some types, but not all.

Reply:
This `Float::Primitive` overload was indeed added as a breaking change in 1.9 because it needs to return `nil`